src/main/java/org/mule/extension/email/api/predicate/IMAPEmailPredicateBuilder.java

```diff
@@ -60,9 +88,74 @@
   @Optional(defaultValue = "INCLUDE")
   private EmailFilterPolicy recent;
 
+  private java.util.Optional<SearchTerm> remoteSearchTerm = empty();
+
+  private final static Logger LOGGER = LoggerFactory.getLogger(IMAPEmailPredicateBuilder.class);
+
+  public void initializeRemoteSearchTerm() {
```

Comment:
I would move all the seach logic to a new class, it can even receive an IMAP email predicate builder to be initialized.